### ISSUE TITLE
Cannot serve an image inline

### ISSUE THREAD

**Issue by fabienlem** [dustin10/VichUploaderBundle#955]
|    Q        |   A
|------------ | -----
| Version     | 1.8.6
| Symfony   | 4.2.2

### Support Question

Hello,

I don't succeed to make the inline display working for an image file.

I reproduced this sample from the documentation: https://github.com/dustin10/VichUploaderBundle/blob/master/Resources/doc/downloads/serving_files_with_a_controller.md#displaying-inline

```
class StorageController extends AbstractController
{
    /**
     * @Route("/product/{id}/image", name="serve_product_image")
     */
    public function serveProductImage(Product $product, DownloadHandler $downloadHandler)
    {
        return $downloadHandler->downloadObject($product, 'imageFile', null, null, false);
    }
}
```

The image file is always downloaded and cannot be displayed inside a <img src=""> tag.

Can you point out what could be wrong?

Thank you!

**Comment by garak** [comment]
Well. that method is actually meant to download an image.

**Comment by fabienlem** [comment]
@garak not according to the documentation that say: "_Instead of forcing the file to be downloaded by you browser, you can set it as an inline content._"

**Comment by garak** [comment]
You're right, my bad.
So, we need some more context

**Comment by fabienlem** [comment]
No problem! What kind of additionnal information can I provide you with?
By the way, thank you for your quick replies!

**Comment by garak** [comment]
You should provide a project where it's possible to reproduce your issue

**Comment by Kern046** [comment]
This is an issue with the response Content-Type. The bundle tries to read a ``mimeType`` property in ``DownloadHandler`` methods and set the content-type to ``application/octet-stream`` if the property is not found.

I tried to add a ``mimeType`` field with hardcoded ``image/jpeg`` value to my picture entity but it's still ``application/octet-stream`` in the response Content-Type header.

I added a mimeType field to my entity, which I populate during the upload, and then I set this value as the Content-Type header. The image was correctly displayed. Here is the code :

```php
$response = $downloadHandler->downloadObject($picture, 'file', null, null, false);
$response->headers->set('Content-Type', $picture->getMimeType());
return $response;
```
I think the real issue here is the mimeType retrieval in the DownloadHandler.
